File: Code/JavaScript/client/src/messages/GreetingMessage.js
import LastAbortMessage from './LastAbortMessage';
import CoreDumpSummaryMessage from './CoreDumpSummaryMessage';
export default class GreetingMessage
{
 static toJSON(o){ 
    const r = {};
   r["tpe"]="g";
    const n = (v)=>v!==undefined&&v!==null;
    if(n(o.lastAbortMessage))
        r["la"]=LastAbortMessage.toJSON(o.lastAbortMessage);
    if(n(o.coreDumpSummaryMessage))
        r["cd"]=CoreDumpSummaryMessage.toJSON(o.coreDumpSummaryMessage);
    return r;
 }
 static fromJSON(o){
    const r = {};
    const n = (v)=>v!==undefined&&v!==null;
    let v;
    v = o["la"];
    if(n(v))
        r.lastAbortMessage=LastAbortMessage.fromJSON(v)
    v = o["cd"];
    if(n(v))
        r.coreDumpSummaryMessage=CoreDumpSummaryMessage.fromJSON(v)
   return r;
 }
}

File: Code/C#/HVPSAPI/Messages/GreetingMessage.cs
using Core.Messages.Messages;
using HVPSAPI.DataMemberNames.Messages;
using HVPSAPI.DataMemberNames.Responses;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace HVPSAPI.Messages
{
    [DataContract]
    public class GreetingMessage : TypedMessageBase
    {
        [JsonPropertyName(GreetingMessageDataMemberNames.LastAbortMessage)]
        [JsonInclude]
        [DataMember(Name = GreetingMessageDataMemberNames.LastAbortMessage)]
        public LastAbortMessage? LastAbortMessage { get; set; }
        [JsonPropertyName(GreetingMessageDataMemberNames.CoreDumpSummaryMessage)]
        [JsonInclude]
        [DataMember(Name = GreetingMessageDataMemberNames.CoreDumpSummaryMessage)]
        public CoreDumpSummaryMessage? CoreDumpSummaryMessage { get; set; }
        public GreetingMessage()
            : base()
        {
            Type = MessageTypes.Greeting;
        }
    }
}
